src/Model/Gallery.php

```diff
@@ -62,51 +62,81 @@ public function __toString()
         return $this->getName() ?? '-';
     }
 
+    /**
```

Comment:
We should not do this for model classes because they are more likely to be customised and overridden.

Reply:
I dont know what kind of customization is needed for a getter and a setter, can you provide an example?

Reply:
Also, if some user needs to customize the methods, he can create a class from the Interfaces, no need to use this Model.

Reply:
Or at least open an issue with an explanation of the needs, and we can still remove the final if we find it's justified.

It's easier to add final now because adding final is not BC but removing final is.

Reply:
e.g. you want to update the `updatedAt` or `enabled` state when setting some property.

> Also, if some user needs to customize the methods, he can create a class from the Interfaces, no need to use this Model.

Yes you could create some custom class, but that's not the idea of reusable models. With the same logic we could make the entity classes final.

Reply:
You could do that on the prePersist or preUpdate (which are not final on purpose)

Reply:
As @VincentLanglet , what about making them final and if someone complains on a concrete use case, we can change it because it will be BC. For your example, using the prePersist could solve the issue (and if you really need to modify all the methods, you might as well create from the interface directly). wdyt?

Reply:
By the way we should make the decision consistent between the Classification and the Media bundle about entities

Reply:
IMO it should be as we did for any other final TBH, it is final until it can be proven that there is a real need that cannot be solved in other ways.

When checking for final, we should ensure we added it for classes and for abstract classes.

That being said, the more time we spend on this, the more we delay releases and working on other bundles...

Reply:
> IMO it should be as we did for any other final TBH, it is final until it can be proven that there is a real need that cannot be solved in other ways.
> 
> When checking for final, we should ensure we added it for classes and for abstract classes.

I agree

Reply:
Still not a fan, but feel to merge this if I'm the only one with this considerations.